src/http/oauth2.ts

```diff
@@ -103,7 +110,7 @@ export function createOAuth2HttpClient(
     ) {
       originalRequest.isRetryWithRefreshedTokens = true;
       await refreshTokens();
-      return http(originalRequest);
+      return httpWithAuth(originalRequest);
```

Comment:
This actually solved a bug introduced in the current version we're working on. Was caught after correcting the test suite, so was definitely worth it already 😬